### DISCUSSION TITLE
Simple Text node

### DISCUSSION THREAD

**Discussion by zeffii** [nortikin/sverchok#4482] [Ideas]
this can either be a "mode" of the String tools node, or a separate node entirely.
![image](https://user-images.githubusercontent.com/619340/169458671-61521f44-8261-4533-9e4c-9e806ede98be.png)

**Comment by zeffii** [comment]
```python
import bpy
import sverchok
from sverchok.node_tree import SverchCustomTreeNode

class SvSimpleText(bpy.types.Node, SverchCustomTreeNode):
    """
    Triggers: Text 
    Tooltip: simple text
    """
    bl_idname = 'SvSimpleText'
    bl_label = 'Simple Text'
    bl_icon = 'FILE_TEXT'
    
    def sv_init(self, context):
        item = self.outputs.new('SvTextSocket', "Text")
        item.custom_draw = "draw_output"
    
    def draw_output(self, socket, context, layout):
        row = layout.row(align=True)
        row.prop(socket, "default_property", text="")

    def process(self):
        text_socket = self.outputs[0]
        text_socket.sv_set([[text_socket.default_property]])
    

classes = [SvSimpleText]
register, unregister = bpy.utils.register_classes_factory(classes)

if __name__ == "__main__":
    if hasattr(bpy.types, "SvSimpleText"):
        bpy.utils.unregister(SvSimpleText)

    register()
```

**Comment by vicdoval** [comment]
I have been using the "Note" node for this use, It is probably to big and a bit strange that you see the text twice but it works
![image](https://user-images.githubusercontent.com/10011941/169483072-35648de1-39cb-476b-88b3-e0fe5bf6ea4b.png)

**Reply by zeffii** [reply]
yeah, i don't like it :)

**Reply by zeffii** [reply]
there's a possiblity to make "simple text" a mode of String tools. (but i won't if you object..)

![image](https://user-images.githubusercontent.com/619340/169485308-3362df91-162b-4630-9766-99f32990ba0b.png)

but i'd like to see the enum dropdown as the F icon only,  - in particular for that mode, so it can all be displayed on one row.

**Comment by zeffii** [comment]
![image](https://user-images.githubusercontent.com/619340/169488057-1e0398f4-93a0-4825-971e-ddc9d4f57970.png)

**Reply by zeffii** [reply]
in any other mode, the node will look as you intended originally.

**Reply by zeffii** [reply]
not sure if I caught all possible control-flow-issues yet..
![image](https://user-images.githubusercontent.com/619340/169489100-103d88b4-642e-4b44-ad7c-6626c2a5e808.png)

**Reply by zeffii** [reply]
https://github.com/nortikin/sverchok/pull/4484

**Reply by vicdoval** [reply]
If you manage to do what you propose it is good for me

**Reply by zeffii** [reply]
it works but i don't like how it introduced extra complexity into the String Tools node.  I think it will be a separate node.

**Reply by zeffii** [reply]
![image](https://user-images.githubusercontent.com/619340/169515003-bb510280-686b-407d-a7a4-af1d8a633e56.png)
